cluster-autoscaler/processors/nodeinfosprovider/mixed_nodeinfos_processor.go

```diff
@@ -36,17 +36,31 @@ import (
 
 const stabilizationDelay = 1 * time.Minute
 
+type cacheItem struct {
+	*schedulerframework.NodeInfo
+	added time.Time
+}
+
 // MixedTemplateNodeInfoProvider build nodeInfos from the cluster's nodes and node groups.
 type MixedTemplateNodeInfoProvider struct {
-	nodeInfoCache map[string]*schedulerframework.NodeInfo
+	nodeInfoCache map[string]cacheItem
+	ttl           *time.Duration
 }
 
 // NewMixedTemplateNodeInfoProvider returns a NodeInfoProvider processor building
 // NodeInfos from real-world nodes when available, otherwise from node groups templates.
-func NewMixedTemplateNodeInfoProvider() *MixedTemplateNodeInfoProvider {
+func NewMixedTemplateNodeInfoProvider(ttl *time.Duration) *MixedTemplateNodeInfoProvider {
 	return &MixedTemplateNodeInfoProvider{
-		nodeInfoCache: make(map[string]*schedulerframework.NodeInfo),
+		nodeInfoCache: make(map[string]cacheItem),
+		ttl:           ttl,
+	}
+}
+
+func (p *MixedTemplateNodeInfoProvider) isCacheItemExpired(added time.Time) bool {
+	if p.ttl == nil {
```

Comment:
nit: Why do you use a pointer at all? If `ttl` was a regular value you wouldn't have to do this.

Reply:
I left pointer for test that mocks MixedTemplateNodeInfoProvider, so there is no need to pass a specific parameter there. However it should be handled during provider creation.

Reply:
So you have different state for tests and a different one for actual execution. If you used a const like `veryLongTime = 123456 * time.Hour` in the tests, you wouldn't need special casing `nil`.

Reply:
I wouldn't say that state for test is different, a nil pointer correspond to default value — the behavior is identical as if --node-info-cache-expire-time flag is not set. https://github.com/kubernetes/autoscaler/pull/4669/files#diff-ad9d6ea8aa127e0dc3b271f9b5761df7ba18f9fd63ffd0a7f69a6886c86ef317R54
I think that if new test is introduced that doesn't test nodeInforCache directly but use it, then the writer shouldn't come up with a new veryLongTime variable. 
Please, explain if my line of thought is wrong. 
(Also, after your argument I used nil pointer for initialising in DefaultProcessor(), which is used not only for test.)